File: Java 15/Matching Specific String.java
import java.io.*;
import java.util.regex.*;

public class Solution {
    public static void main(String[] args) throws IOException {
        String test = new BufferedReader(new InputStreamReader(System.in)).readLine().trim();
        Matcher m = Pattern.compile("hackerrank").matcher(test);
        long count = m.results().count();
        System.out.println("Number of matches : " + count);
    }
}

File: C++20/Matching Specific String.cpp
#include <iostream>
#include <regex>
#include <string>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    string s;
    getline(cin, s);

    regex pattern("hackerrank");
    auto begin = sregex_iterator(s.begin(), s.end(), pattern);
    auto end = sregex_iterator();
    int count = distance(begin, end);
    cout << "Number of matches : " << count;

    return 0;
}